### DISCUSSION TITLE
Database create(table:) - Why should multiple columns in primary key be consecutive?

### DISCUSSION THREAD

**Discussion by changyjacky** [groue/GRDB.swift#1326] [Q&A] [answered]
Sqlite allow this CREATE TABLE statement:
``` sql
CREATE TABLE team_member
(  playerid INTEGER,
   note TEXT,
   teamid INTEGER,
   PRIMARY KEY (playerid, teamid)
)
``` 
But GRDB Database create table function must be:
``` swift
try db.create(table: "team_member") { t in
    t.primaryKey {
        t.column("playerid", .integer)
        t.column("teamid", .integer)
    }
    t.column("note", .text)
}
```
You just **CAN'T** keep the "note" column  between the two primary-key columns.
It sounds not that bad; But when you design the UI, it will be a trouble,
you must change the column order when user take non-consecutive columns as primary key.
It **doen't make sense**. 

Is there any alternative solution? 
(Just don't want to turn back to SQL CREATE statement...)

**Comment by groue** [comment, marked as answer]
Hello @changyjacky,

Your alternative solution is to add an explicit PRIMARY KEY constraint:

```swift
try db.create(table: "team_member") { t in
    t.column("playerid", .integer).notNull()
    t.column("note", .text)
    t.column("teamid", .integer).notNull()
    t.primaryKey(["playerId", "teamId"])
}
```

Take care, though, to add a NOT NULL constraint to each primary key column.

I'll update the [documentation](https://swiftpackageindex.com/groue/grdb.swift/documentation/grdb/database/create(table:options:body:)) so that it is clearer.

**Reply by groue** [reply]
Documentation has been updated in 302fc69f3aa634c9478ea73e1ba87b1c18b0738e

**Reply by changyjacky** [reply]
> Documentation has been updated in [302fc69](https://github.com/groue/GRDB.swift/commit/302fc69f3aa634c9478ea73e1ba87b1c18b0738e)

Thanks a lot!

**Reply by BarnoTD** [reply]
I have a question, will this syntax work too?

```swift
try db.create(table: "team_member") { t in
    t.column("playerid", .integer).primaryKey()
    t.column("teamid", .integer).primaryKey()
    t.column("note", .text)
}
```

**Reply by groue** [reply]
Hi @BarnoTD. Did you try running your code before asking? I expect that SQLite will throw an error complaining that you define multiple primary keys. Open your Terminal and see by yourself:

```
% sqlite3
sqlite> CREATE TABLE team_member(
          playerid INTEGER PRIMARY KEY,
          teamid INTEGER PRIMARY KEY,
          note TEXT);
Parse error: table "team_member" has more than one primary key
```

So, no, it will not work.

In your case, follow the [documentation](https://swiftpackageindex.com/groue/grdb.swift/documentation/grdb/databaseschemamodifications) and write:

```swift
try db.create(table: "team_member") { t in
    t.primaryKey {
        t.column("playerid", .integer)
        t.column("teamid", .integer)
    }
    t.column("note", .text)
}
```
